sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequest.java

```diff
@@ -78,13 +79,16 @@ public static RntbdRequest decode(final ByteBuf in) {
     void encode(final ByteBuf out) {
 
         final int expectedLength = RntbdRequestFrame.LENGTH + this.headers.computeLength();
-        final int start = out.readerIndex();
+        final int start = out.writerIndex();
```

Comment:
could you please explain the scenario which this bug shows itself?

why do we need this?

Reply:
This change was originally found and fixed in PR https://github.com/Azure/azure-sdk-for-java/pull/14697/files#diff-cdd54a91cbc899239eec06998abac39eL81. 
We have not came across any issue today since the writerIndex and readerIndex should be the same. But the writerIndex is the more correct value we should use.  

But also feel it may be a risk to depend on the underlying ByteBuf implementation details.

Reply:
I just read the whole code. makes sense to use writerIndex. thanks